sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/DirectoryAPITests.groovy

```diff
@@ -392,23 +396,78 @@ class DirectoryAPITests extends APISpec {
         }
         primaryDirectoryClient.createFile(dirPrefix + 2, 1024)
         for (int i = 0; i < 3; i++) {
-            nameList.add(dirPrefix + i)
+            nameList << (dirPrefix + i)
         }
 
         when:
-        def fileRefIter = primaryDirectoryClient.listFilesAndDirectories(namer.getResourcePrefix() + extraPrefix, maxResults, null, null).iterator()
+        def returnedFileList = primaryDirectoryClient.listFilesAndDirectories(
+            new ShareDirectoryListFilesAndDirectoriesOptions()
+                .setPrefix(namer.getResourcePrefix() + extraPrefix)
+                .setMaxResultsPerPage(maxResults).setShareFileTraits(traits)
+                .setIncludeExtendedInfo(includeExtendedInfo),
+            null, null).collect()
 
         then:
-        for (int i = 0; i < numOfResults; i++) {
-            Objects.equals(nameList.pop(), fileRefIter.next().getName())
+        if (expectingResults) {
+            assert nameList == returnedFileList*.getName()
+        }
+        else {
+            assert returnedFileList.size() == 0
         }
-        !fileRefIter.hasNext()
+        true
```

Comment:
why do we have true here?

Reply:
leftover from before an `else` existed so that the block returned *something*. Will remove.